### DISCUSSION TITLE
It would be great if there was haptic feedback.

### DISCUSSION THREAD

**Discussion by BrettDean** [rxhanson/Rectangle#1365] [Ideas]
It would be great if there was haptic feedback.

**Comment by rxhanson** [comment]
Thanks for the feedback. When would you want haptic feedback? When a window footprint is displayed or changes?

**Reply by BrettDean** [reply]
Exactly, adding haptic feedback when a window footprint is displayed or changes would be a very interesting experience.  :)

**Reply by rxhanson** [reply]
Would you still want this even if the haptic feedback is rather weak? Last I worked on haptic feedback (which is in two of my apps, [Multitouch](https://multitouch.app) and [Charmstone](https://charmstone.app)), the haptic feedback I can add in as a 3rd party developer is almost unnoticeable. In order to get stronger haptic feedback, it appeared that you had to resort to private framework usage (something I wouldn't consider).

**Reply by BrettDean** [reply]
I tried the haptic feedback in [Multitouch](https://multitouch.app/), and actually, it felt okay. This type of vibration seems sufficient for Rectangle.

**Comment by rodhash** [comment]
I see it was implemented and I enabled its checkbox but, are we talking about vibration response on Mac or what kind of response I was supposed to see?

I use Mac mini and not a laptop so is there any sound response that I was supposed to  ~~see~~  hear? After enabling its checkbox I don't see any options to configure the response type/level/etc.

**Reply by BrettDean** [reply]
Maybe try Magic Trackpad? I'm not using a Mac mini, and I can feel vibration response on MacBook.

**Reply by Aaron-Rumpler** [reply]
The haptic feedback is only for Apple trackpads. They have a haptic engine in them that simulates clicks (quite realistically) and is also used for other haptic feedback (e.g. snapping).

**Comment by BrettDean** [comment]
The Haptic feedback feels like it's missing one vibration. Just like a switch. When the mouse drags the window close to the screen edge, it flips the switch on, and when the mouse drags the window away from the screen edge, it flips the switch off.

**Reply by SkyrilHD** [reply]
I just saw this discussion post and I personally disagree with it as, when you drag a window to the edge of the screen, the vibration implies that the window has an impact to the edge. Therefore, there should be no vibration when dragging away.

What I described is also in Apple's Human Interface Guidelines: https://developer.apple.com/design/human-interface-guidelines/playing-haptics#Impact

> Impact haptics provide a physical metaphor you can use to complement a visual experience. For example, **people might feel a tap when a view snaps into place** or a thud when two heavy objects collide.
